src/main/groovy/org/transmartproject/batch/biodata/BioMarkerDictionary.groovy

```diff
@@ -0,0 +1,35 @@
+package org.transmartproject.batch.biodata
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Component
+import org.transmartproject.batch.clinical.db.objects.Tables
+
+/**
+ * Access bio markers dictionary.
+ */
+@Component
+class BioMarkerDictionary {
+
+    @Lazy
+    Map<String, String> uniprotIdToUniprotName = fetchUniprotIdToUniprotName()
+
+    @Autowired
+    NamedParameterJdbcTemplate jdbcTemplate
+
+    String getUniprotNameByUniporotId(String uniprotId) {
+        uniprotIdToUniprotName[uniprotId]
+    }
+
+    private Map<String, String> fetchUniprotIdToUniprotName() {
+        jdbcTemplate.queryForList(
+                """SELECT
+                  primary_external_id,
+                  bio_marker_name
+                 FROM ${Tables.BIO_MARKER}
+                 WHERE bio_marker_type = :marker_type""",
+                [marker_type: 'PROTEIN'])
+                .collectEntries { [it.primary_external_id, it.bio_marker_name] }
+    }
```

Comment:
I'd prefer this loading to be done in a separate step to avoid the first chunk taking a lot more time to process that the following ones; the user can be confused with the wait. Also, make it job scoped so that the memory can be reclaimed after the job finishes (not really relevant now, but it can be in the future).